mlflow/utils/__init__.py

```diff
@@ -120,3 +119,37 @@ def _get_fully_qualified_class_name(obj):
     Obtains the fully qualified class name of the given object.
     """
     return obj.__class__.__module__ + "." + obj.__class__.__name__
+
+
+def _inspect_original_var_name(var):
+    import inspect
+    try:
+        original_var_name = None
+
+        frame = inspect.currentframe().f_back
+        while frame is not None:
+            arg_info = inspect.getargvalues(frame)
+
+            fixed_args = [arg_info.locals[arg_name] for arg_name in arg_info.args]
+            varlen_args = list(arg_info.locals[arg_info.varargs]) if arg_info.varargs else []
+            keyword_args = list(arg_info.locals[arg_info.keywords].values()) if arg_info.keywords else []
+
+            all_args = fixed_args + varlen_args + keyword_args
+
+            # check whether `var` is in arg list first. If yes, go to check parent frame.
+            if any(var is arg for arg in all_args):
+                # the var is passed in from caller, check parent frame.
+                frame = frame.f_back
+                continue
+
+            for var_name, var_val in frame.f_locals.items():
+                if var_val is var:
+                    original_var_name = var_name
+                    break
+
+            break
+
+        return original_var_name
+
+    except BaseException:
```

Comment:
Is this here for testing purposes? What should we return in the event of an exception - `None`? Can we catch `Exception` instead of `BaseException` so that we don't catch things like `KeyboardInterrupt`, which might prevent users from terminating a program that uses MLflow?

Reply:
> What should we return in the event of an exception

If raise exception, we return None, "None" means fallback to use f'dataset_{id(dataset)}' as the dataset name.